enforcer/src/main/java/org/wso2/micro/gateway/enforcer/util/TLSUtils.java

```diff
@@ -0,0 +1,140 @@
+    private static final String X509 = "X.509";
+
+    /**
+     * Read the certificate file and return the certificate.
+     *
+     * @param filePath Filepath of the corresponding certificate
+     * @return Certificate
+     */
+    public static Certificate getCertificateFromFile(String filePath)
+            throws CertificateException, IOException {
+        return getCertsFromFile(filePath, true).get(0);
+    }
+
+    public static Certificate getCertificateFromString(String certificateContent)
+            throws CertificateException, IOException {
+        // A single certificate file is expected
+        try (InputStream inputStream = new ByteArrayInputStream(certificateContent.getBytes())) {
+            CertificateFactory fact = CertificateFactory.getInstance(X509);
+            return fact.generateCertificate(inputStream);
+        }
+    }
+
+    /**
+     * Add the certificates to the the truststore.
+     *
+     * @param filePath Filepath of the corresponding certificate or directory containing the certificates
+     * @return Certificate
+     */
+    public static void addCertsToTruststore(KeyStore trustStore, String filePath) throws IOException {
+        if (!Files.exists(Paths.get(filePath))) {
+            log.error("The provided certificates directory/file path does not exist. : " + filePath);
+            return;
+        }
+        if (Files.isDirectory(Paths.get(filePath))) {
+            log.debug("Provided Path is a directory: " + filePath);
+            Files.walk(Paths.get(filePath)).filter(path -> {
+                Path fileName = path.getFileName();
+                return fileName != null && (fileName.toString().endsWith(".crt") ||
+                        fileName.toString().endsWith(".pem"));
+            }).forEach(path -> {
+                try {
+                    List<Certificate> certificateList = getCertsFromFile(path.toAbsolutePath().toString(), false);
+                    certificateList.forEach(certificate -> {
+                        try {
+                            trustStore.setCertificateEntry(RandomStringUtils.random(10, true, false),
+                                    certificate);
+                        } catch (KeyStoreException e) {
+                            log.error("Error while adding the trusted certificates to the trustStore.", e);
+                        }
+                    });
+                    log.debug("Certificate Added to the truststore : " + path.toAbsolutePath());
+                } catch (CertificateException | IOException e) {
+                    log.error("Error while adding certificates to the truststore.", e);
+                }
+            });
+        } else {
+            log.debug("Provided Path is a regular File Path : " + filePath);
+            try {
+                List<Certificate> certificateList = getCertsFromFile(filePath, false);
+                certificateList.forEach(certificate -> {
```

Comment:
Move the code to a common function as it is shared.

Reply:
fixed with https://github.com/wso2/product-microgateway/pull/1582/commits/b804e40a426ab4a0c8f06cc8c5d68598fbfc8882